src/XrdCms/XrdCmsRRQ.hh

```diff
@@ -117,8 +117,20 @@ void *Respond();
 
 struct Info
       {
-        Info(): Add2Q(0), PBack(0), Resp(0), Multi(0), luFast(0), luSlow(0),
-                rdFast(0), rdSlow(0) {}
+        Info()
+	{Reset();}
+	
+	void Reset()
+	{Add2Q  = 0;
+	 PBack  = 0;
+	 Resp   = 0;
+	 Multi  = 0;
+	 luFast = 0;
+	 luSlow = 0;
+	 rdFast = 0;
+	 rdSlow = 0;
+	}
+
```

Comment:
What is the compiler complaining about here that makes you change all inline initialization to a function call? I am concerned that this structure may defeat compiler optimization.

Reply:
The reason I add the 'Reset' method is here:

https://github.com/xrootd/xrootd/commit/60da79db805d122a7190535b8dead0f306e5a2f9#diff-bffded318632dc8e33cbdb6c4dcc1d34L134

In XrdCmsRRQ.cc the compiler was complaining that an object with a non-trivial constructor is being initialized with memset. As the constructor does basically the same as 'Reset' I though that implementing the constructor in terms of 'Reset' method would simplify the code. IMHO the compiler will be smart enough to first inline 'Reset' and then the constructor. That said, I don't have any strong feelings on this one, I can revert to the old constructor implementation if you wish (as long as the 'Reset' stays).